test/e2e-helm/e2e_helm_project_test.go

```diff
@@ -0,0 +1,187 @@
+				Expect(err).NotTo(HaveOccurred())
+				return logOutput
+			}
+			Eventually(managerContainerLogs, time.Minute, time.Second).Should(ContainSubstring("Installed release"))
+
+			By("Getting the release name")
+			releaseName, err := tc.Kubectl.Get(
+				false,
+				tc.Kind, "-o", "jsonpath={..status.deployedRelease.name}")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(releaseName)).NotTo(BeIdenticalTo(0))
+
+			By("Checking the release(CR) deployment status")
+			verifyReleaseUp := func() string {
+				output, err := tc.Kubectl.Command(
+					"rollout", "status", "deployment", releaseName)
+				Expect(err).NotTo(HaveOccurred())
+				return output
+			}
+			Eventually(verifyReleaseUp, time.Minute, time.Second).Should(ContainSubstring("successfully rolled out"))
+
+			By("Ensuring the created Service for the release(CR)")
+			crServiceName, err := tc.Kubectl.Get(
+				false,
+				"Service", "-l", fmt.Sprintf("app.kubernetes.io/instance=%s", releaseName),
+				"-o", "jsonpath={..metadata.name}")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(crServiceName)).NotTo(BeIdenticalTo(0))
+
+			By("Scaling deployment replicas to 2")
+			_, err = tc.Kubectl.Command(
+				"scale", "deployment", releaseName, "--replicas", "2")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("verifying the deployment automatically scales back down to 1")
+			verifyRelease := func() error {
+				replicas, err := tc.Kubectl.Get(
+					false,
+					"deployment", releaseName, "-o", "jsonpath={..spec.replicas}")
+				Expect(err).NotTo(HaveOccurred())
+				if replicas != "1" {
+					return fmt.Errorf("release(CR) deployment with %s replicas", replicas)
+				}
+				return nil
+			}
+			Eventually(verifyRelease, time.Minute, time.Second).Should(Succeed())
+
+			By("Updating replicaCount to 2 in the CR manifest")
+			testutils.ReplaceInFile(filepath.Join(tc.Dir, sampleFile), "replicaCount: 1", "replicaCount: 2")
+
+			By("Applying CR manifest with replicaCount: 2")
+			_, err = tc.Kubectl.Apply(false, "-f", sampleFile)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking Deployment replicas spec is equals 2")
+			verifyReleaseUpgrade := func() error {
+				replicas, err := tc.Kubectl.Get(
+					false,
+					"deployment", releaseName, "-o", "jsonpath={..spec.replicas}")
+				Expect(err).NotTo(HaveOccurred())
+				if replicas != "2" {
+					return fmt.Errorf("release(CR) deployment with %s replicas", replicas)
+				}
+				return nil
+			}
+			Eventually(verifyReleaseUpgrade, time.Minute, time.Second).Should(Succeed())
+
+			By("Ensuring the CR gets reconciled and the release was Upgraded")
+			managerContainerLogsAfterUpdateCR := func() string {
+				logOutput, err := tc.Kubectl.Logs(controllerPodName, "-c", "manager")
+				Expect(err).NotTo(HaveOccurred())
+				return logOutput
+			}
+			Eventually(managerContainerLogsAfterUpdateCR, time.Minute, time.Second).Should(ContainSubstring("Upgraded release"))
```

Comment:
Logically, it probably makes more sense to do this test right after patching the CR and before checking the deployment replicas.